### PR TITLE
Enable transferring funds to ENS address

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-sdk/lib/core/services/TransferService.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 export class TransferService {
   constructor(private deployedWallet: DeployedWallet) {}
 
   async transfer(transferDetails: TransferDetails) {
-    ensure(isProperAddress(transferDetails.to), InvalidAddress, transferDetails.to);
+    ensure(isProperAddress(transferDetails.to) || isValidEnsName(transferDetails.to), InvalidAddressOrEnsName, transferDetails.to);
     ensureNotNull(this.deployedWallet, WalletNotFound);
-    const message = encodeTransferToMessage({...transferDetails, from: this.deployedWallet.contractAddress});
+    let message;
+    if (isProperAddress(transferDetails.to)) {
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/core/services/TransferService.ts]: Proposal: extract function `getTargetAddress(SDK, addressOrEnsName)`

File: universal-login-sdk/lib/core/utils/getTargetAddress.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import UniversalLoginSDK from '../..';
+import {isProperAddress, ensure} from '@universal-login/commons';
+import {InvalidAddressOrEnsName} from './errors';
+
+export const getTargetAddress = async (sdk: UniversalLoginSDK, addressOrEnsName: string) => {
+  if(isProperAddress(addressOrEnsName)) {
```
Comment (rzadp) [inline universal-login-sdk/lib/core/utils/getTargetAddress.ts]:
```suggestion
  if (isProperAddress(addressOrEnsName)) {
```

File: universal-login-sdk/lib/core/utils/errors.ts [hunk 3]
```diff
@@ -67,10 +67,10 @@ export class ValidationFailed extends SDKError {
   }
 }
 
-export class InvalidAddress extends ValidationFailed {
+export class InvalidAddressOrEnsName extends ValidationFailed {
   constructor(address: string) {
-    super(`Address ${address} is not valid`, 'InvalidAddress');
-    Object.setPrototypeOf(this, InvalidAddress.prototype);
+    super(`${address} is not valid`, 'InvalidAddressOrEnsName');
```
Comment (xitronix) [inline universal-login-sdk/lib/core/utils/errors.ts]: Consider rename address variable (recipient maybe, but i am not sure)

File: universal-login-commons/lib/core/utils/sanitizers/asTransferDetails.ts [hunk 4]
```diff
@@ -1,10 +1,9 @@
 import {asObject, asString, Sanitizer} from '@restless/sanitizers';
 import {TransferDetails} from '../../..';
-import {asEthAddress} from '@restless/ethereum';
 import {asGasParameters} from './asGasParameters';
 
 export const asTransferDetails: Sanitizer<TransferDetails> = asObject<TransferDetails>({
-  to: asEthAddress,
+  to: asString,
```
Comment (xitronix) [inline universal-login-commons/lib/core/utils/sanitizers/asTransferDetails.ts]: what do you think about introduce new sanitizer? (asRecipient)

Comment (rzadp) [reply universal-login-commons/lib/core/utils/sanitizers/asTransferDetails.ts]: +1, possibly in follow up

File: universal-login-sdk/lib/core/utils/errors.ts (hunk 3 above)
Comment (rzadp) [reply universal-login-sdk/lib/core/utils/errors.ts]: or `addressOrEnsName`

File: universal-login-sdk/lib/core/utils/errors.ts [hunk 5]
```diff
@@ -67,10 +67,10 @@ export class ValidationFailed extends SDKError {
   }
 }
 
-export class InvalidAddress extends ValidationFailed {
+export class InvalidAddressOrEnsName extends ValidationFailed {
   constructor(address: string) {
```
Comment (rzadp) [inline universal-login-sdk/lib/core/utils/errors.ts]:
```suggestion
  constructor(addressOrEnsName: string) {
```

File: universal-login-sdk/lib/core/utils/errors.ts (hunk 3 above)
Comment (rzadp) [inline universal-login-sdk/lib/core/utils/errors.ts]:
```suggestion
    super(`${addressOrEnsName} is not valid`, 'InvalidAddressOrEnsName');
```

File: universal-login-sdk/lib/core/utils/errors.ts [hunk 6]
```diff
@@ -67,10 +67,10 @@ export class ValidationFailed extends SDKError {
   }
 }
 
-export class InvalidAddress extends ValidationFailed {
-  constructor(address: string) {
-    super(`Address ${address} is not valid`, 'InvalidAddress');
-    Object.setPrototypeOf(this, InvalidAddress.prototype);
+export class InvalidAddressOrEnsName extends ValidationFailed {
+   constructor(addressOrEnsName: string) {
```
Comment (rzadp) [inline universal-login-sdk/lib/core/utils/errors.ts]:
```suggestion
  constructor(addressOrEnsName: string) {
```

File: universal-login-sdk/lib/core/utils/errors.ts (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/lib/core/utils/errors.ts]: Done

File: universal-login-commons/lib/core/utils/sanitizers/asTransferDetails.ts (hunk 4 above)
Comment (JustynaBroniszewska) [reply universal-login-commons/lib/core/utils/sanitizers/asTransferDetails.ts]: Let's do it in nex pr. This one is big enough.

**Pull request merged**